rcl/test/rcl/test_events.cpp

```diff
@@ -565,3 +591,143 @@ TEST_F(CLASSNAME(TestEventFixture, RMW_IMPLEMENTATION), test_pubsub_liveliness_k
   // clean up
   tear_down_publisher_subscriber();
 }
+
+/*
+ * Basic test of publisher and subscriber incompatible qos callback events.
+ * Only implemented in opensplice at the moment.
+ */
+TEST_P(TestEventFixture, test_pubsub_incompatible_qos)
+{
+  if (!is_opensplice) {
+    GTEST_SKIP();
+  }
+
+  const auto & input = GetParam();
+  const auto & qos_policy_kind = std::get<0>(input);
+  const auto & publisher_qos_profile = std::get<1>(input);
+  const auto & subscription_qos_profile = std::get<2>(input);
+  const auto & error_msg = std::get<3>(input);
+  setup_publisher_and_subscriber(
+    RCL_PUBLISHER_OFFERED_INCOMPATIBLE_QOS,
+    publisher_qos_profile,
+    RCL_SUBSCRIPTION_REQUESTED_INCOMPATIBLE_QOS,
+    subscription_qos_profile);
+
+  rcl_ret_t ret;
+
+  // try to receive event, retry for 10s
+  static const size_t max_iterations = 1000;
+  static const auto wait_period = 10ms;
+  bool pub_event_success = false;
+  bool sub_event_success = false;
+  for (size_t i = 0; i < max_iterations; ++i) {
```

Comment:
can `conditional_wait_for_msgs_and_events` be reused somehow?

Reply:
I have refactored this test case to use `conditional_wait_for_msgs_and_events()`.